bigtable/cmd/cbt/cbt_test.go

```diff
@@ -171,3 +178,438 @@ func TestParseColumnsFilter(t *testing.T) {
+			continue
+		}
+		if len(fams) != len(tc.oFams) ||
+			len(cols) != len(tc.oCols) ||
+			len(line) != len(tc.nextLine) {
+			t.Errorf("parseCsvHeaders() did not fail, incorrect output sizes found, fams: %d, cols:%d, line:%d", len(fams), len(cols), len(line))
+			continue
+		}
+		for i, f := range fams {
+			if f != tc.oFams[i] {
+				t.Errorf("Incorrect column families idx:%d, got: %q, want %q", i, fams[i], tc.oFams[i])
+				continue
+			}
+		}
+		for i, c := range cols {
+			if c != tc.oCols[i] {
+				t.Errorf("parseImportArgs() did not fail for column names idx:%d, got: %q, want %q", i, cols[i], tc.oCols[i])
+				continue
+			}
+		}
+		for i, v := range line {
+			if v != tc.nextLine[i] {
+				t.Errorf("parseImportArgs() did not fail for next line idx:%d, got: %q, want %q", i, cols[i], tc.oCols[i])
+				continue
+			}
+		}
+	}
+}
+
+func setupEmulator(t *testing.T, tables, families []string) (context.Context, *bigtable.Client) {
+	srv, err := bttest.NewServer("localhost:0")
+	if err != nil {
+		t.Fatalf("Error starting bttest server: %s", err)
+	}
+
+	ctx := context.Background()
+
+	conn, err := grpc.Dial(srv.Addr, grpc.WithInsecure())
+	if err != nil {
+		t.Fatalf("Error %s", err)
+	}
+
+	proj, instance := "proj", "instance"
+	adminClient, err := bigtable.NewAdminClient(ctx, proj, instance, option.WithGRPCConn(conn))
+	if err != nil {
+		t.Fatalf("Error %s", err)
+	}
+
+	for _, ta := range tables {
+		if err = adminClient.CreateTable(ctx, ta); err != nil {
+			t.Fatalf("Error %s", err)
+		}
+		for _, f := range families {
+			if err = adminClient.CreateColumnFamily(ctx, ta, f); err != nil {
+				t.Fatalf("Error %s", err)
+			}
+		}
+	}
+
+	client, err := bigtable.NewClient(ctx, proj, instance, option.WithGRPCConn(conn))
+	if err != nil {
+		t.Fatalf("Error %s", err)
+	}
+
+	return ctx, client
+}
+
+func TestCsvParseAndWrite(t *testing.T) {
+	ctx, client := setupEmulator(t, []string{"my-table"}, []string{"my-family", "my-family-2"})
+
+	tbl := client.Open("my-table")
+	fams := []string{"", "my-family", "my-family-2"}
+	cols := []string{"", "col-1", "col-2"}
+	rowData := [][]string{
+		{"rk-0", "A", "B"},
+		{"rk-1", "", "C"},
+	}
+	matchCount := 3
+
+	byteData, err := writeAsCSV(rowData)
+	if err != nil {
+		t.Fatal(err)
+	}
+	reader := csv.NewReader(bytes.NewReader(byteData))
+
+	sr := safeReader{r: reader}
+	if err = sr.parseAndWrite(ctx, tbl, fams, cols, 1, 1, 1); err != nil {
+		t.Fatalf("parseAndWrite() failed unexpectedly")
+	}
+
+	// vaildate table entries
+	colMap := make(map[string][]string)
+	for i := range fams {
+		var col []string
+		for _, r := range rowData {
+			col = append(col, r[i])
+		}
+		colMap[fams[i]+":"+cols[i]] = col
+	}
+	for rowIdx, data := range rowData {
+		if row, err := tbl.ReadRow(ctx, data[0]); err != nil {
+			t.Errorf("Error %s", err)
+		} else {
+			for _, fam := range fams {
+				for _, column := range row[fam] {
+					colId := string(column.Column)
```

Comment:
Seems the linter is upset about this var. in Go, Id => ID.

```
bigtable/cmd/cbt/cbt_test.go:410:6: var colId should be colID
bigtable/cmd/cbt/cbt_test.go:482:5: var colId should be colID
bigtable/cmd/cbt/cbt_test.go:597:7: var colId should be colID
```

Reply:
ok, I have a cleaner way to do this verification logic too, will update